src/APL/ioredis/ioredis-apl.ts

```diff
@@ -0,0 +1,239 @@
+import { SpanKind, SpanStatusCode } from "@opentelemetry/api";
+import { SemanticAttributes } from "@opentelemetry/semantic-conventions";
+import Redis from "ioredis";
+
+import { getOtelTracer, OTEL_APL_SERVICE_NAME } from "../../open-telemetry";
+import { APL, AplConfiguredResult, AplReadyResult, AuthData } from "../apl";
+import { createAPLDebug } from "../apl-debug";
+
+/**
+ * Configuration options for IORedisAPL
+ */
+type IORedisAPLConfig = {
+  /** Redis client instance to use for storage */
+  client: Redis;
+  /** Optional key to use for the hash collection. Defaults to "saleor_app_auth" */
+  hashCollectionKey?: string;
+};
+
+/**
+ * Redis implementation of the Auth Persistence Layer (APL).
+ * This class provides Redis-based storage for Saleor App authentication data.
```

Comment:
```suggestion
 * This class provides IORedis-based storage for Saleor App authentication data.
```

Can we also add link to IORedis docs here? If I will be using this in my app it will be awesome to have link in JSDoc 😄